admin-manual/installation-setup/upgrading/upgrading.rst

```diff
@@ -47,40 +46,99 @@ upgrade again.
+    sudo service elasticsearch restart
+    sleep 60s
+    # Configure the ES backup
+    curl -XPUT "localhost:9200/_snapshot/backup-repo" -H 'Content-Type: application/json' -d \
+    '{
+         "type": "fs",
+         "settings": {
+         "location": "./",
+         "compress": true
+         }
+     }'
+    # Take the actual backup, and copy it to a safe place
+    curl -X PUT "localhost:9200/_snapshot/backup-repo/am_indexes_backup?wait_for_completion=true"
+    cp /var/lib/elasticsearch/backup-repo elasticsearch-backup -rf
+
+
+
+For more info, refer to the `ElasticSearch 1.7 docs <https://www.elastic.co/guide/en/elasticsearch/reference/1.7/modules-snapshots.html>`_
+
+.. _upgrade-search-indices:
+
+Upgrade Elasticsearch and search indexes
+----------------------------------------
+
+.. note::
+
+   Ignore this section if you are planning to run Archivematica without search
+   indexes. Instead, follow the instructions on :ref:`how to upgrade
+   Archivematica in indexless mode <upgrade-indexless>`.
+
+Archivematica |release| uses Elasticsearch 6.x as its search engine. If you're
+upgrading from Archivematica 1.8.x or lower, where Elasticsearch 1.x was the
+supported version, it's required to upgrade your Elasticsearch cluster and
+indexes to the new version.
+
+To complete this upgrade it is important to know if you have access to your
+transfer backlog and AIP storage locations in the local filesystem.
+These are usually located in the following paths:
+
+* :file:`/var/archivematica/sharedDirectory/www/AIPsStore/transferBacklog`
+* :file:`/var/archivematica/sharedDirectory/www/AIPsStore`
+
+You should confirm the paths of your installation in the Locations tab of the
+Storage Service.
+
+If you have access to those locations, the recommended method for the upgrade is
+to :ref:`recreate the indexes <recreate-indexes>`. Otherwise, you'll need to
+:ref:`reindex from another cluster <cluster-reindex>`.
+
 
 .. _upgrade-ubuntu:
 
-Upgrade on Ubuntu
------------------
+Upgrade on Ubuntu packages
+--------------------------
 
-Before you start: Archivematica 1.7 stopped using ``/usr/share/python`` as the
-home for our Python virtual environments. We did this because it was causing
-Ubuntu Xenial to break when attempting to upgrade a system package called
-``python-minimal``. If you're running Ubuntu Xenial the following commands are
-going to be necessary to continue the upgrade:
+1. If you choose the :ref:`recreate the indexes <recreate-indexes>` , ElasticSearch 1.7 
+needs to be removed before proceeding with the upgrade. This can be done with:
 
 .. code:: bash
 
-   sudo rm -rf /usr/share/python/archivematica-dashboard
-   sudo rm -rf /usr/share/python/archivematica-mcp-server
-   sudo rm -rf /usr/share/python/archivematica-mcp-client
+   sudo apt-get remove --purge elasticsearch
+   sudo mv /var/lib/elasticsearch /var/lib/elasticsearch-1.7.5
+   sudo mv /etc/elasticsearch /etc/elasticsearch-1.7.5
```

Comment:
This ordered list might end up being misnumbered because this code block is interrupting the list, but I'm not 100% sure. If it is, the solution is to move the code block, including the directive, over by 3 spaces so that it's in line with the beginning of the list item text. That way, it's considered part of the list item and will not interrupt the numbering.

```
1. Here is my list item text.

   .. code:: bash
      sudo rm -rf /usr/share/python/archivematica-dashboard
```